JPS_VIRTUALSENSOR/AermodAgent/src/main/java/com/cmclinnovations/aermod/AermodAgent.java

```diff
@@ -222,10 +218,30 @@ public void processRequestParameters(DerivationInputs derivationInputs, Derivati
         gdalClient.uploadVectorStringToPostGIS(EnvConfig.DATABASE, elevationLayerName, geoJSON2.toString(),
                 new Ogr2OgrOptions(), true);
 
-        for (int i = 0; i < dispLayerNames.size(); i++) {
+        for (int i = 0; i < receptorHeights.size(); i++) {
             double height = receptorHeights.get(i);
             JSONObject geoJSON = aermod.getGeoJSON(EnvConfig.PYTHON_SERVICE_URL, outputFileURL, srid, height);
-            String dispLayerName = dispLayerNames.get(i);
+            // The derivation IRI, pollutant ID, simulation time and height properties are
+            // added here to
+            // enable the
+            // post-processing agent to retrieve the relevant contours for the
+            // visualization.
+            JSONArray features = geoJSON.getJSONArray("features");
+            for (int j = 0; j < features.length(); j++) {
+                JSONObject feature = features.getJSONObject(i);
+                JSONObject featureProperties = feature.getJSONObject("properties");
+                featureProperties.put("derivationIRI", derivationInputs.getDerivationIRI());
+                featureProperties.put("pollutantID", EnvConfig.POLLUTANT_ID);
+                featureProperties.put("simulation_time", simulationTime);
+                featureProperties.put("height", height);
+                feature.put("properties", featureProperties);
+                features.put(j, feature);
```

Comment:
these two lines might be redundant

Reply:
I believe this line creates a new object:

JSONObject featureProperties = feature.getJSONObject("properties");

So if I modify featureProperties, don't I need to update the original object in feature accordingly?

Reply:
I believe the methods modify the objects directly, rather than creating a new object in memory